lib/Dialect/FIRRTL/FIRRTLReductions.cpp

```diff
@@ -423,12 +423,12 @@ struct MemoryStubber : public OpReduction<firrtl::MemOp> {
       auto result = memOp.getResult(i);
       auto name = builder.getStringAttr(Twine(memOp.getName()) + "_" +
                                         memOp.getPortNameStr(i));
-      auto wire =
-          builder
-              .create<firrtl::WireOp>(result.getType(), name,
-                                      firrtl::NameKindEnum::DroppableName,
-                                      memOp.getPortAnnotation(i), StringAttr{})
-              .getResult();
+      auto wire = builder
+                      .create<firrtl::WireOp>(
+                          cast<firrtl::FIRRTLType>(result.getType()), name,
```

Comment:
Wondering if memories can have non-FIRRTLType results, and if not whether we can narrow them in ODS accordingly?

Basically any time code confidently does a "cast" at one point -- hard to not wonder if that was known why it isn't that way already.  Same for all these casts.

Reply:
Yes, memories should be narrowed if they aren't already (I thought they were).

The casts tend to be from using the op/value general accessors rather than op-specific ones.

Reply:
Memory data types are passive base types, but the ports are not.

Reply:
Okay thanks!